core/src/main/scala/filodb.core/memstore/PartKeyLuceneIndex.scala

```diff
@@ -240,7 +241,26 @@ class PartKeyLuceneIndex(ref: DatasetRef,
   private val utf8ToStrCache = concurrentCache[UTF8Str, String](PartKeyLuceneIndex.MAX_STR_INTERN_ENTRIES)
 
   //scalastyle:off
-  private val searcherManager = new SearcherManager(indexWriter, null)
+  private val searcherManager =
+    if (disableIndexCaching) {
+      new SearcherManager(indexWriter,
+        new SearcherFactory() {
+          override def newSearcher(reader: IndexReader, previousReader: IndexReader): IndexSearcher = {
+            val indexSearcher = super.newSearcher(reader, previousReader)
+            indexSearcher.setQueryCache(null)
```

Comment:
do we need to still set the cache to null if shouldCache returns false ?

Reply:
Describe why it is detrimental too

Reply:
Good question @sandeep6189 Initially I set the caching policy and disable caching anything but later found setting cache to null also disables cache (at least from unit tests) and also eliminates the check for whether to cache for not. However, I left both changes in. 

Hello @whizkido good to see your comments  :)

When we profile a FIloDB instance with a large index and which sees a lot of repeating queries, we see the following flame graph (as taken by async profiler)

<img width="1975" alt="Screenshot 2024-02-02 at 10 00 10 AM" src="https://github.com/filodb/FiloDB/assets/1028064/961739ab-51c4-4065-82db-b3932dde02db">

We  see almost 80% stack traces are related to index searches and about 50% of the are related to caching.  Also [this](https://www.mail-archive.com/java-user@lucene.apache.org/msg51649.html) archive mentions disabling caching which may or may not work in our case   (also they mention disabling cache in a different way than what this PR does).  The idea is to have that knob to let us disable the caching and then profile again. It might be worse but we can measure and check.